homeassistant/components/intellifire/sensor.py

```diff
@@ -0,0 +1,128 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+from intellifire4py import IntellifirePollData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.dt import utcnow
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN
+
+
+class IntellifireSensor(CoordinatorEntity, SensorEntity):
```

Comment:
In a follow-up PR, I think that you could create a base class in `entity.py` that would be inherited by both Sensor and BinarySensor platforms:
```python
class IntellifireEntity(CoordinatorEntity):
    """Define a generic class for Intellifire entities."""

    coordinator: IntellifireDataUpdateCoordinator
    _attr_attribution = "Data provided by unpublished Intellifire API"

    def __init__(
        self,
        coordinator: IntellifireDataUpdateCoordinator,
        description: EntityDescription,
    ) -> None:
```